### PR TITLE
Replace while-loop with C++ find_if function

### PR DESCRIPTION
**Pull request by med-ayssar**

### CODE REVIEW

File: nestkernel/universal_data_logger.h [hunk 1]
```diff
@@ -241,13 +243,12 @@ nest::UniversalDataLogger< HostNode >::connect_logging_device( const DataLogging
...
-  size_t j = 0;
-  while ( j < n_loggers and data_loggers_[ j ].get_mm_node_id() != mm_node_id )
-  {
-    ++j;
-  }
-  if ( j < n_loggers )
+
+  auto item = std::find_if( data_loggers_.begin(),
+    data_loggers_.end(),
+    [ &mm_node_id ]( const DataLogger_& dl ) { return dl.get_mm_node_id() == mm_node_id; } );
```
Comment (jougs) [inline nestkernel/universal_data_logger.h:249]: I think there's no need to repeat the name here. `[ & ]` captures all variables used in the lambda by reference.

```suggestion
    [ & ]( const DataLogger_& dl ) { return dl.get_mm_node_id() == mm_node_id; } );
```

File: nestkernel/universal_data_logger.h [hunk 2]
```diff
@@ -24,8 +24,10 @@
 #define UNIVERSAL_DATA_LOGGER_H
 
 // C++ includes:
+#include <algorithm>
 #include <vector>
 
+
```
Comment (jougs) [inline nestkernel/universal_data_logger.h:30]:
```suggestion
```

**Review by jougs — COMMENTED** [review]
Just some minor cleanup.

**Review by jougs — APPROVED** [review]
:+1:

File: nestkernel/universal_data_logger.h [hunk 3]
```diff
@@ -241,13 +242,12 @@ nest::UniversalDataLogger< HostNode >::connect_logging_device( const DataLogging
...
   const index mm_node_id = req.get_sender().get_node_id();
-  const size_t n_loggers = data_loggers_.size();
-  size_t j = 0;
-  while ( j < n_loggers and data_loggers_[ j ].get_mm_node_id() != mm_node_id )
-  {
-    ++j;
-  }
-  if ( j < n_loggers )
+
+  auto item = std::find_if( data_loggers_.begin(),
```
Comment (hakonsbm) [inline nestkernel/universal_data_logger.h:246]:
```suggestion
  const auto item = std::find_if( data_loggers_.begin(),
```

**Review by hakonsbm — CHANGES_REQUESTED** [review]

**Review by hakonsbm — APPROVED** [review]
Thanks!

**Pull request merged**